src/pages/instances/StoragePoolSelectTable.tsx

```diff
@@ -0,0 +1,129 @@
+import { FC } from "react";
+import { Button, MainTable } from "@canonical/react-components";
+import ScrollableTable from "components/ScrollableTable";
+import StoragePoolSize from "pages/storage/StoragePoolSize";
+import { LxdStoragePool } from "types/storage";
+
+interface Props {
+  pools: LxdStoragePool[];
+  onSelect: (pool: string) => void;
+  disablePool?: (pool: LxdStoragePool) => string | undefined;
```

Comment:
that was my initial approach, however since this is a more general component, we should consider what if we want to disable multiple rows?  We could pass in a list of objects with your suggested structure, but I think it would be easier to allow the definition of a disable function since through the callback arguments we can get all the info we need for a storage pool wdyt?

Reply:
I was thinking about this too and as for the two use cases at hand we have only one row to disable, probably fine to only support that case for now. If we ever need more, we can enhance it.

Reply:
cool, done :+1: